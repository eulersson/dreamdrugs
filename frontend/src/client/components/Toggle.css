#Toggle {
  cursor: pointer;
  position: relative;
  float: right;

  width: 30px;

  margin: 15px 15px auto -53px;
  padding: 10px 4px;

  border-radius: 5px;
  background-color: #369;

  transition: 1s;
  visibility: visible;
}

#Toggle.toggled {
  border-radius: 19px;
  background-color: white;
}

#bar1, #bar2, #bar3 {
    width: 30px;
    height: 3px;
    border-radius: 1px;
    background-color: white;
    transition: 0.4s;
}

#bar1, #bar2 {
    margin-bottom: 3px;
}

.toggled #bar1, .toggled #bar2, .toggled #bar3 {
  height: 4px;
  background-color: #369;
}

.toggled #bar1 {
  transform: translateX(-7px) rotate(-45deg) translate(0px, 10px);
}

.toggled #bar2 {
  opacity: 0;
}

.toggled #bar3 {
  transform: translate(-7px) rotate(45deg) translate(0px, -10px);
}
