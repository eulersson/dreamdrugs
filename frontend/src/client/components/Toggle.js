import React from "react";
import PropTypes from "prop-types";

import "./Toggle.css";

function Toggle(props) {
  return (
    <div
      id="Toggle"
      className={props.switch ? "toggled" : ""}
      onClick={props.onChange}
    >
      <div id="bar1" />
      <div id="bar2" />
      <div id="bar3" />
    </div>
  );
}

Toggle.propTypes = {
  switch: PropTypes.bool.isRequired,
  onChange: PropTypes.func.isRequired
};

export default Toggle;
