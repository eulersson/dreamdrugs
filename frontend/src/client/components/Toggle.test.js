import React from "react";
import Toggle from "./Toggle";

import renderer from 'react-test-renderer';

import { shallow } from "enzyme";

describe("toggle", () => {
  it("renders correctly", () => {
    const component = renderer
      .create(<Toggle switch={true} onChange={() => {}} />)
      .toJSON();
    expect(component).toMatchSnapshot();
  });

  it("has thee bars", () => {
    const component = shallow(<Toggle />);
    expect(component.closest("div").children()).toHaveLength(3);
  });
});
