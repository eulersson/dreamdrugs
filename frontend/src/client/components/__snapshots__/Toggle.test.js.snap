// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`toggle renders correctly 1`] = `
<div
  className="toggled"
  id="Toggle"
  onClick={[Function]}
>
  <div
    id="bar1"
  />
  <div
    id="bar2"
  />
  <div
    id="bar3"
  />
</div>
`;
